src/core/utils/helpers/general.ts

```diff
@@ -149,3 +149,37 @@ export const getRenewableMaterials = (list: LoanV2[]) => {
 export const getAmountOfRenewableLoans = (list: LoanV2[]) => {
   return getRenewableMaterials(list).length;
 };
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+export const groupBy = (arr, property) =>
+  arr.reduce(
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    (acc, curr) => {
+      const propertyValue = curr[property];
+      if (propertyValue in acc) {
+        return { ...acc, [propertyValue]: acc[propertyValue].concat(curr) };
+      }
+      return { ...acc, [propertyValue]: [curr] };
+    },
+    {} // This empty object is cache
+  );
+
+//  TODO:  make a typeScript function that can be used insted of the groupBy
+//
+// export const groupObjectArrayByProperty = <T>(arr: T[], property: string) =>
+//   arr.reduce<Record<string, T[]>>(
+//     (acc: { [key: string]: T[] }, curr: T) => {
+//       const propertyValue = curr[property as keyof T];
+//       if (!propertyValue) {
+//         return acc;
+//       }
+//       if (propertyValue in acc) {
+//         return { ...acc, [propertyValue]: [...acc[propertyValue], curr] };
+//       }
+//       return { ...acc, [propertyValue]: [curr] };
+//     },
+//     {} as Record<string, T[]> | Record<string, unknown>
+//     // This empty object is cache
+//   );
```

Comment:
```suggestion
export const groupObjectArrayByProperty = <
  P extends string,
  T extends Record<P, string> & Record<string, unknown>
>(
  arr: T[],
  property: P
) =>
  arr.reduce<Record<string, T[]>>(
    (result: { [key: string]: T[] }, current: T) => {
      const groupBy = current[property];
      // Merge into result if the property already exist.
      if (groupBy in result) {
        return {
          ...result,
          [groupBy]: [...result[groupBy], current]
        };
      }
      // Otherwise create new property.
      return { ...result, [groupBy]: [current] };
    },
    {}
  );
```